examples/35_gemm_softmax/softmax_epilogue.hpp

```diff
@@ -0,0 +1,300 @@
+    auto [m_coord, n_coord, k_coord, l_coord] = blk_coord_mnkl;
+    Tensor gC = gC_mnl(_,_,m_coord,n_coord,l_coord);                                                 // (BLK_M,BLK_N)
+    Tensor gD = gD_mnl(_,_,m_coord,n_coord,l_coord);                                                 // (BLK_M,BLK_N)
+    Tensor gMax = gMax_mnl(_,m_coord,n_coord,l_coord);
+    Tensor gSum = gSum_mnl(_,m_coord,n_coord,l_coord);
+
+    //Represent the shared tensor
+    Tensor sC = make_tensor(make_smem_ptr(reinterpret_cast<ElementAccumulator*>(smem_buf)), make_layout(make_shape(M_tile, N_tile)));
+
+    // Partition the tiles to match the accumulator partitioning
+    auto thr_mma = tiled_mma.get_thread_slice(thread_idx);
+    Tensor tCgD = thr_mma.partition_C(gD);                                       // (VEC,THR_M,THR_N)
+    Tensor tCgC = thr_mma.partition_C(gC);                                       // (VEC,THR_M,THR_N)
+    Tensor tCsC = thr_mma.partition_C(sC);                                       // (VEC,THR_M,THR_N)
+
+    static_assert(is_static<FrgLayout>::value, "Accumulator layout must be static");
+    CUTE_STATIC_ASSERT_V(size(tCgC) == size(tCgD),
+        "Source and destination must have the same number of elements.");
+    CUTE_STATIC_ASSERT_V(size(tCgD) == size(accumulators),
+        "Accumulator count must have the same destination element count.");
+
+    // Make an identity coordinate tensor for predicating our output MN tile
+    auto cD = make_identity_tensor(make_shape(unwrap(shape<0>(gD)), unwrap(shape<1>(gD))));
+    Tensor tCcD = thr_mma.partition_C(cD);
+
+    if(is_source_needed()){
+      CUTLASS_PRAGMA_UNROLL
+      for (int i = 0; i < size<0>(accumulators); ++i) {
+        CUTLASS_PRAGMA_UNROLL
+        for (int j = 0; j < size<1>(accumulators); ++j) {
+          CUTLASS_PRAGMA_UNROLL
+          for (int k = 0; k < size<2>(accumulators); ++k) {
+            if (elem_less(tCcD(i,j,k), make_coord(get<0>(residue_mnk), get<1>(residue_mnk)))) {
+              accumulators(i,j,k) = epilogue_op(accumulators(i,j,k), tCgC(i,j,k));
+              tCgD(i,j,k) = accumulators(i,j,k);
+              tCsC(i,j,k) = accumulators(i,j,k);
+            }
+          }
+        }
+      }
+    } else{
+      CUTLASS_PRAGMA_UNROLL
+      for (int i = 0; i < size<0>(accumulators); ++i) {
+        CUTLASS_PRAGMA_UNROLL
+        for (int j = 0; j < size<1>(accumulators); ++j) {
+          CUTLASS_PRAGMA_UNROLL
+          for (int k = 0; k < size<2>(accumulators); ++k) {
+            if (elem_less(tCcD(i,j,k), make_coord(get<0>(residue_mnk), get<1>(residue_mnk)))) {
+              accumulators(i,j,k) = epilogue_op(accumulators(i,j,k));
+              tCgD(i,j,k) = accumulators(i,j,k);
+              tCsC(i,j,k) = accumulators(i,j,k);
+            } 
+          }
+        }
+      }
+    }
+
+    syncthreads();
+
+    // assumption for reductions: size<0>(sC) == block size
+    assert(size<0>(sC) == BlockDimX() * BlockDimy() * BlockDimZ());
```

Comment:
Isn't there a typo here @t4c1 "BlockDimy -> BlockDimY" ?

Reply:
Yes, opened a new PR to fix it: https://github.com/codeplaysoftware/cutlass-fork/pull/176